### ISSUE TITLE
Check validation of progress structure before getting to Summary

### ISSUE THREAD

**Issue by nmadruga** [openmsupply/application-manager-web-app#48]
Should not allow getting to the summary page before all fields are valid.

**Comment by CarlosNZ** [comment]
Also need to run validation again for all questions on Summary page load (to catch any earlier questions that might be invalidated by later answers). Not sure if this can be done as part of this issue -- if not, can whoever is doing this issue please create a new issue for that?

**Comment by nmadruga** [comment]
@CarlosNZ 
> Also need to run validation again for all questions on Summary page load (to catch any earlier questions that might be invalidated by later answers). Not sure if this can be done as part of this issue -- if not, can whoever is doing this issue please create a new issue for that?

After we have chatted about the difference with this new idea of only allowing the user to go to the summary page when all required elements have valid responses, this should be fine. We have 2 issues now to deal with it (before we can do this one): 
#194 - To recalculate all responses validity when one is changed
#193 - To redirect user to the invalid response when they try accessing the summary page 

> I'm thinking that they probably should be done together, as the result of the validation check (on load) can be used to inform the submission check (so it doesn't have to recompute validations on submit, it just checks the result of the on-load validation check).

Yes. I think that's what will be done now. When we are loading the Summary, the code will check if all elements are valid, and otherwise send the user to the page with the invalid element, if not will load. Then the user can Submit.
